### PR TITLE
Adds Berserk

### PR DESCRIPTION
**Pull request by Neerti**
Adds a new modifier that makes someone big, strong, and see everything sans hud elements in red. After awhile they get tired and get a debuff.
Can obtain this modifier from various sources. See the [modifier file](https://github.com/Neerti/Polaris/blob/01b7277cc3eb801b0b8fd0b198fde4493bc99592/code/modules/mob/_modifiers/modifiers_misc.dm) for details.
Rage
Gif (warning: long): https://puu.sh/yKCsi/84b7c21a44.gif

### CODE REVIEW

**Comment by Magneld** [comment]
Rip and Tear.

**Comment by Anewbe** [comment]
YOU HAVE HUGE GUTS

File: code/game/gamemodes/changeling/powers/enrage.dm [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	set desc = "Causes you to go Berserk."
+
+	var/datum/changeling/changeling = changeling_power(30,0,100)
+	if(!changeling)
+		return 0
+
+	var/modifier_to_use = /datum/modifier/berserk/changeling
+	if(src.mind.changeling.recursive_enhancement)
+		modifier_to_use = /datum/modifier/berserk/changeling/recursive
+		to_chat(src, "<span class='notice'>We optimize our levels of anger, which will avoid excessive stress on ourselves.</span>")
```
Comment (Anewbe) [inline code/game/gamemodes/changeling/powers/enrage.dm]: Not a complaint, but "We optimize our levels of anger" made me laugh real hard.

File: code/modules/mob/_modifiers/modifiers_misc.dm [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	on_expired_text = "<span class='notice'>The blaze of rage inside you has ran out.</span>"
+	stacks = MODIFIER_STACK_EXTEND
+
+	// The good stuff.
+	slowdown = -1							// Move a bit faster.
+	attack_speed_percent = 0.66				// Attack at 2/3 the normal delay.
+	outgoing_melee_damage_percent = 1.5		// 50% more damage from melee.
+	max_health_percent = 1.5				// More health as a buffer, however the holder might fall into crit after this expires if they're mortally wounded.
+	disable_duration_percent = 0.25			// Disables only last 25% as long.
+	icon_scale_percent = 1.2				// Look scarier.
```
Comment (Anewbe) [inline code/modules/mob/_modifiers/modifiers_misc.dm]: Not sure this is desirable, but it won't stop me from merging this.
They shrink back to normal when it ends, yes?

File: code/modules/xenobio/items/extracts.dm [hunk 3]
```diff
@@ -564,12 +564,19 @@
...
-			to_chat(S, "<span class='warning'>An intense wave of rage almost overcomes you, but you remain in control of yourself.</span>")
+			to_chat(S, "<span class='warning'>An intense wave of rage is felt from inside, but you remain in control of yourself.</span>")
 			continue
 
 		S.enrage()
 
+	for(var/mob/living/carbon/human/H in view(get_turf(holder.my_atom)))
+		if(H.species.name == "Promethean")
+			H.add_modifier(/datum/modifier/berserk, 30 SECONDS)
+			to_chat(H, "<span class='warning'>An intense wave of rage is felt from inside, but you remain in control of yourself.</span>")
```
Comment (Anewbe) [inline code/modules/xenobio/items/extracts.dm]: It says they remain in control, but they've got the same Berserk modifier, is that intended?

File: code/modules/mob/_modifiers/modifiers_misc.dm (hunk 2 above)
Comment (Neerti) [reply code/modules/mob/_modifiers/modifiers_misc.dm]: Yes.

File: code/modules/xenobio/items/extracts.dm (hunk 3 above)
Comment (Neerti) [reply code/modules/xenobio/items/extracts.dm]: Meant to show that they still have free will and such and don't go 'rabid' as their slime counterparts might.

Comment (Anewbe) [reply code/modules/xenobio/items/extracts.dm]: Alrighty.

**Pull request merged**